core/vm/evm.go

```diff
@@ -56,11 +58,13 @@ func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, err
 			precompiles = PrecompiledContractsHomestead
 		}
 		if p := precompiles[*contract.CodeAddr]; p != nil {
-			switch p.(type) {
-			case *XDCxEpochPrice:
-				p.(*XDCxEpochPrice).SetTradingState(evm.tradingStateDB)
-			case *XDCxLastPrice:
-				p.(*XDCxLastPrice).SetTradingState(evm.tradingStateDB)
+			if evm.chainConfig.IsTIPXDCXMiner(evm.BlockNumber) {
+				switch p.(type) {
```

Comment:
should use `p := p.(type)` to simplify the code.

Reply:
sorry, I didn't get this part
p is used in later lines

Reply:
```go
switch p := p.(type) {
			case *XDCxEpochPrice:
				p.SetTradingState(evm.tradingStateDB)
			case *XDCxLastPrice:
				p.SetTradingState(evm.tradingStateDB)
			}
```

Reply:
thanks! fixed